cmd/bootkube/render.go

```diff
@@ -99,9 +99,6 @@ func validateRenderOpts(cmd *cobra.Command, args []string) error {
 	if (renderOpts.etcdCAPath != "" || renderOpts.etcdCertificatePath != "" || renderOpts.etcdPrivateKeyPath != "") && (renderOpts.etcdCAPath == "" || renderOpts.etcdCertificatePath == "" || renderOpts.etcdPrivateKeyPath == "") {
 		return errors.New("You must specify either all or none of --etcd-ca-path, --etcd-certificate-path, and --etcd-private-key-path")
 	}
-	if renderOpts.etcdCertificatePath != "" && renderOpts.selfHostedEtcd {
```

Comment:
Actually it seems like you want to keep this check. AFAICT you are ignoring the etcd certificate flags when using self-hosted and just generating certs from the master CA.

Maybe add some more information to the flag or this error that explains what the behavior is for self-hosted etcd + TLS.

Also need to edit line 214 to not display that warning when using self-hosted etcd.